bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/AbstractCloudBigtableTableDoFn.java

```diff
@@ -34,7 +39,41 @@
 public abstract class AbstractCloudBigtableTableDoFn<In, Out> extends DoFn<In, Out> {
   private static final long serialVersionUID = 1L;
 
-  private static final CloudBigtableConnectionPool pool = new CloudBigtableConnectionPool();
+  static final CloudBigtableConnectionPool pool = new CloudBigtableConnectionPool();
+
+  public static void logException(
+      Logger log, String context, RetriesExhaustedWithDetailsException exception) {
+    log.warn("For context {}: exception occured during bulk writing: {}", context,
+      exception.getExhaustiveDescription());
+    log.error("For context {}: First bulk writing exception", exception.getCause(0), context);
```

Comment:
Why log both warning and error?
If this is specific for bulk write, name it as such. logErrorOnBulkWrite()?